src/ParametersHandler/tests/ParametersHandlerTest.cpp

```diff
@@ -0,0 +1,138 @@
+                      << parameterName << " is different from the one expected" << std::endl;
+            return false;
+        }
+
+        return true;
+    }
+
+    std::unique_ptr<IParametersHandler<BasicImplementation>> getGroup(const std::string& name) const
+    {
+        auto group = m_map.find(name);
+        if (group == m_map.end())
+            return std::make_unique<BasicImplementation>();
+
+        std::unordered_map<std::string, std::any> map;
+        try
+        {
+            map = std::any_cast<decltype(map)>(group->second);
+        } catch (const std::bad_any_cast& exception)
+        {
+            std::cerr << "[BasicImplementation::getGroup] The element named " << name
+                      << " is not a 'std::unordered_map<std::string, std::any>'" << std::endl;
+            return std::make_unique<BasicImplementation>();
+        }
+
+        return std::make_unique<BasicImplementation>(map);
+    }
+
+    friend std::ostream& operator<<(std::ostream& os, const BasicImplementation& handler)
+    {
+        std::string key;
+        for (const auto parameters: handler.m_map)
+            key += parameters.first + " ";
+
+        return os << key;
+    }
+
+    ~BasicImplementation() = default;
+};
+
+TEST_CASE("Get parameters")
+{
+    BasicImplementation::unique_ptr parameterHandler = std::make_unique<BasicImplementation>();
+    parameterHandler->setParameter("answer_to_the_ultimate_question_of_life", 42);
+    parameterHandler->setParameter("pi", 3.14);
+    parameterHandler->setParameter("Fibonacci Numbers", std::vector<int>{1, 1, 2, 3, 5, 8, 13, 21});
+    parameterHandler->setParameter("John", std::string("Smith"));
+
+    SECTION("Get integer")
+    {
+        int element;
+        REQUIRE(parameterHandler->getParameter("answer_to_the_ultimate_question_of_life", element));
+        REQUIRE(element == 42);
+    }
+
+    SECTION("Get Double")
+    {
+        double element;
+        REQUIRE(parameterHandler->getParameter("pi", element));
+        REQUIRE(element == 3.14);
+    }
+
+    SECTION("Get String")
+    {
+        std::string element;
+        REQUIRE(parameterHandler->getParameter("John", element));
+        REQUIRE(element == "Smith");
+    }
+
+    SECTION("Get Vector")
+    {
+        std::vector<int> element;
+        REQUIRE(parameterHandler->getParameter("Fibonacci Numbers", element));
+        REQUIRE(element == std::vector<int>{1, 1, 2, 3, 5, 8, 13, 21});
+    }
+
+    SECTION("Get Group")
+    {
+        BasicImplementation::unique_ptr groupHandler = parameterHandler->getGroup("CARTOONS");
+        groupHandler->setParameter("Donald's nephews",
+                                   std::vector<std::string>{"Huey", "Dewey", "Louie"});
+        std::vector<std::string> element;
+        REQUIRE(groupHandler->getParameter("Donald's nephews", element));
+        REQUIRE(element == std::vector<std::string>{"Huey", "Dewey", "Louie"});
+    }
```

Comment:
Can you test also the print function as in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/13/files#diff-2187c40c7243c514d7eca0ce41d286eeR72-R75?

Reply:
Done in 987775d4217f2cd1dcc3951b6453b1c7526b44d2